docs/proposals/build-skipping.adoc

```diff
@@ -0,0 +1,69 @@
+
+However if one has multiple builds in one repo, the pipeline build runs all build tasks even for those areas where nothing has changed.
+
+A simple scenario is a repo with a backend and frontend component whose build outputs are combined by a subsequent docker build in a subsequent deploy task.
+
+There is no good reason when only the frontend code changes that the backend build is redone as well and vice versa of course.
+
+== Solution
+
+All tasks shall be enabled to support build skipping.
+
+The following is added to the images of these tasks:
+
+- `git`
+
+Task parameter
+
+- `cache-output-dir` a boolean defaulting to `'true'`.
+
+The solution builds on the caching mechanism introduced in issue #147.
+
+`ods-start` supports a new area in the cache for the purpose of caching build task outputs.
+This area is below folder `.ods-cache/build-task`.
+
+If `cache-output-dir` is `'true'` build tasks shall copy their build outputs to `.ods-cache/build-task/$(git-sha-working-dir)`, where
+
+- `git-sha-working-dir` is the git sha of the working directory which is determined as follows: `git-sha-working-dir=$(git rev-parse "HEAD:$working_dir")`  (https://stackoverflow.com/a/58600111).
+
+However if `cache-output-dir` is `'true'` and there is prior build output at that location, cached files are made available in the tasks `output-dir`. The cache and output dir will be on the same PVC so that hard linking can be utilized if needed.
+
+For traceability the task result sets a parameter `build-reused`` to 'true' (or perhaps the git sha). +
+**Question:** as far as I know `ods-finish` expects all build task reports to be available. However one might rather receive a reference to the original build reports instead. This would allow to not keep these reports in the cache. Can ods-finish be changed so that reports of reused builds are not required to be provided again?
+
+If either `cache-output-dir` is not `'true'` or prior build output is not in the cache the tasks builds as usual and provides the build output to the `output-dir`.
+
+Finally if a build happened and `cache-output-dir` is `'true'` the task makes the build output available in the the cache. Again this should be speedy as hard linking should enable avoiding to copy these files. In addition this can likely be implemented out side of the individual task build scripts.
+
+**To be determined **: If feasible: script(s) or go programs could be provided that implements the mentioned logic and would be available in all tasks images.
+
+=== Cleanup
+
+`ods-start` is changed to delete prior build output from the cache if it is older than 7 days (if configurable 7 days would be the default).
+
+== Pro
+
+* Build skipping!
+
+* Flag `cache-output-dir` allows to control build skipping independent of other global caching.
+
+* Build skipping should have little penalty if no reuse actually occurs.
+
+* The task build scripts can (perhaps) be unaware of build skipping.
+
+== Con
```

Comment:
I guess one thing to explore would be how this feature works for builds where the workspace is the root of a repo in the case of merge commits. I believe that build skipping wouldn't work, and that is fine for me, but maybe worth mentioning here?